### DISCUSSION TITLE
I need to implement OTA, Not able to download file from azure blob

### DISCUSSION THREAD

**Discussion by ijaznoufal** [nanoframework/Home#1366] [General]
_httpClient.DefaultRequestHeaders.Add("x-ms-blob-type", "BlockBlob");
 HttpResponseMessage response = _httpClient.Get("[https://xdatas.blob.core.windows.net/ota/test.pe"](https://tinkerbloxdatas.blob.core.windows.net/ota/test.txt%22));
 response.EnsureSuccessStatusCode();
 var responseBody = response.Content.ReadAsString();

for _httpClient.HttpsAuthentCert = new X509Certificate()// iam using same from https://github.com/BWalti/nf-ssl/blob/main/nf-ssl/Resources/AzureRootCerts.pem

but this is the error iam getting 
Issue : An unhandled exception of type 'System.Net.Http.HttpRequestException' occurred in System.Net.Http.dll

**Comment by jingqunhu** [comment]
Maybe the pe file is too big

**Comment by jingqunhu** [comment]
You can use WebRequest to subcontract and download the upgrade package
